### PR TITLE
Wanting make 5 letter codes editable only by admin&owner

### PR DESCRIPTION
**Pull request by wantingxu7**
# Description
截屏2023-10-20 下午1 03 00

## Related PRS (if any):
…

## Main changes explained:
- Update WeeklySummariesReport.jsx file for Owner/Admin/whoever is given "edit team code" permission to edit 5-letter team code in 'Weekly Summaries Report page'
- Update UserProfile.jsx file for Owner/Admin/whoever is given "edit team code" permission to edit 5-letter team code in profile's teams tab
…

## How to test:
1. check into current branch
2. do `npm install` and `npm run start:local` to run this PR locally
3. log as owner/admin user
        a. go to Reports -> Weekly Summaries Report
        b. verify that the 5-letter team code is editable
        c. go to User Profile -> Teams tab
        d. verify that the 5-letter team code is editable
4. edit a team code, refresh the page, and check the edited team code is saved
5. log as any role EXCEPT admin/owner(like volunteer, manager, mentor, etc.)
        a. go to Reports -> Weekly Summaries Report
        b. verify that the 5-letter team code is NOT editable
        c. go to User Profile -> Teams tab
        d. verify that the 5-letter team code is NOT editable

## Screenshots or videos of changes:
**editable team code example:**
截屏2023-10-20 下午1 11 24

截屏2023-10-20 下午1 45 17

**uneditable team code example:**
截屏2023-10-20 下午1 17 32

截屏2023-10-20 下午1 21 32

### CODE REVIEW

**Comment by Iris-Song** [comment]
Hi, when I log in as an administrator and try to update 5-letter team code, an alert show that below: 
image
image

**Review by StrawberryCalpico — CHANGES_REQUESTED** [review]
Hi, I tested your PR. When I login as Admin, all team code boxes are blank. I can't see the code. When I login in as Volunteer, there's no team code box. I think there may be some issues with loading team code to the page?
Here is when I login as admin:
Screenshot 2023-10-20 at 9 52 20 PM
here is the view as volunteer role who has no right to edit team code.
Screenshot 2023-10-20 at 9 53 21 PM

**Review by Cloid — CHANGES_REQUESTED** [review]
Hi there.

I tested with the Admin and Volunteer edge cases provided.
Changing team code as Admin gives an error that @Iris-Song  talked about in their screenshot.

With the Volunteer profile, I tried accessing the Weekly Report summary via the dashboard dropdown and page URL. I just get redirected back to the main home screen which I assume is the intended behavior since you shouldn't have those permissions by default.

I gained access to an owner account and adjusted the permission sets to edit team code enabled the following options:
- “See Only Weekly Summary Reports Tab” 
- "Edit Team 4-Digit Codes" 

Logging back into the Volunteer page, I tried to change the team code and got the following error:
Screenshot 2023-10-20 181255

Options I enabled for Volunteer profile:
Screenshot 2023-10-20 181333
Screenshot 2023-10-20 181337

**Review by Shubhu0 — CHANGES_REQUESTED** [review]
Hi @wantingxu7  I tested your PR and it is working but i noticed a few things

the codes are editable as admin and owner, but some teams do not have any codes whatsoever. Dont know if that was a loading issue or it is as it is. But everything else was working as intended.


image
image

**Comment by mounica-dingari-by** [comment]
Hi @wantingxu7 , I've tested your PR and for an Admin/Owner, I can't see any code for Weekly summaries Reports tab, but it's editable and same with the Profile->Teams tab. But when I login as volunteer, the Reports is not visible in the Nav bar and in the Teams tab, the code text box is not visible, which is fine. But , even when logged in as Owner/Admin, I'm unable to find any code for any of the users. Is there any particular branch that I can test from?

Screenshot from 2023-10-20 20-28-42

Screenshot from 2023-10-20 20-22-06
Screenshot from 2023-10-20 20-24-00
Screenshot from 2023-10-20 20-28-37

**Comment by RheaWu1212** [comment]
Hi I tested your PR, log in as Owner --> able to edit codes, log in as volunteer --> popup message, not able to edit code.
Screen Shot 2023-10-21 at 19 05 17
Screen Shot 2023-10-21 at 19 08 26

**Comment by wantingxu7** [comment]
figuring out the mentioned bugs, labeling it "do not review"

**Review by cvtqx — CHANGES_REQUESTED** [review]
Hi @wantingxu7!
When I log in as admin I can see and edit the codes however, the changes are not being saved.

In  reports -> weekly summaries :


In profile -> teams:


When I log in as other than admin/owner, the codes are not editable:

Screen Shot 2023-10-28 at 9 00 05 PM

Screen Shot 2023-10-28 at 9 00 28 PM

Also volunteers are not be able to see reports tab, so perhaps you need to remove volunteer role from your  'how to test' description so noone gets confused.

File: src/components/UserProfile/UserProfile.jsx [hunk 1]
```diff
@@ -15,367 +15,367 @@ import {
...
     const originalProjectsBeingDisplayed = projectsProperties.filter(
       item =>
         JSON.stringify(item) ===
         JSON.stringify(
           originalProjectProperties.filter(elem => elem.projectName === item.projectName)[0],
         ),
-    );
+    )
 
     const compare =
```
Comment (John-Flavian) [inline src/components/UserProfile/UserProfile.jsx:200]: Duplicated work.

File: src/components/UserProfile/UserProfile.jsx [hunk 2]
```diff
@@ -15,367 +15,367 @@ import {
...
+    })
 
     const originalTeamsBeingDisplayed = teamsProperties.filter(
       item =>
         JSON.stringify(item) ===
         JSON.stringify(originalTeamProperties.filter(elem => elem.teamName === item.teamName)[0]),
-    );
+    )
 
     const compare =
```
Comment (John-Flavian) [inline src/components/UserProfile/UserProfile.jsx:166]: There is no need to have repeated work; you can refactor into a function or variable.

**Review by savio-henrique — CHANGES_REQUESTED** [review]
Hi @wantingxu7 , I've tested this PR, I couldn't edit the team codes as I was getting an 403 error, even in the profile page I couldn't find the code that I tried to edit into my team.

# Screenshots

403 error:
ERROR

Team code not showing up, as it was not edited:
no-team

**Pull request closed without merging**